code/modules/surgery/organs/augments_arms.dm

```diff
@@ -334,3 +334,101 @@
 		/obj/item/surgical_drapes,
 		/obj/item/knife/combat/cyborg,
 	)
+
+/obj/item/organ/internal/cyberimp/arm/muscle
+	name = "\proper Strong-Arm empowered musculature implant"
+	desc = "When implanted, this cybernetic implant will enhance the muscles of the arm to deliver more power-per-action."
+	icon_state = "muscle_implant"
+
+	zone = BODY_ZONE_R_ARM
+	slot = ORGAN_SLOT_RIGHT_ARM_AUG
+
+	actions_types = list()
+	
+	///The amount of damage dealt by the empowered attack.
+	var/punch_damage = 13
+	///Will the punch smash people into walls?
+	var/gentle = TRUE
+	///How far away your attack will throw your oponent
+	var/attack_throw_range = 1
+	///Minimum throw power of the attack
+	var/throw_power_min = 1
+	///Maximum throw power of the attack
+	var/throw_power_max = 4
+	///How long will the implant malfunction if it is EMP'd
+	var/emp_base_duration = 9 SECONDS 
+
+/obj/item/organ/internal/cyberimp/arm/muscle/Insert(mob/living/carbon/reciever, special = FALSE, drop_if_replaced = TRUE)
+	. = ..()
+	if(ishuman(reciever)) //Sorry, only humans
+		RegisterSignal(reciever, COMSIG_HUMAN_EARLY_UNARMED_ATTACK, PROC_REF(on_attack_hand))
+
+/obj/item/organ/internal/cyberimp/arm/muscle/Remove(mob/living/carbon/implant_owner, special = 0)
+	. = ..()
+	UnregisterSignal(implant_owner, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+
+/obj/item/organ/internal/cyberimp/arm/muscle/emp_act(severity)
+	. = ..()
+	if((organ_flags & ORGAN_FAILING) || . & EMP_PROTECT_SELF)
+		return
+	owner.balloon_alert(owner, "your arm spasms wildly!")
+	organ_flags |= ORGAN_FAILING
+	addtimer(CALLBACK(src, PROC_REF(reboot)), 90 / severity)
+
+/obj/item/organ/internal/cyberimp/arm/muscle/proc/reboot()
+	organ_flags &= ~ORGAN_FAILING
+	owner.balloon_alert(owner, "your arm stops spasming!")
+
+/obj/item/organ/internal/cyberimp/arm/muscle/proc/on_attack_hand(mob/living/carbon/human/source, atom/target, proximity, modifiers)
+	SIGNAL_HANDLER
+
+	if(source.get_active_hand() != source.get_bodypart(check_zone(zone)) || !proximity)
+		return
+	if(!source.combat_mode || LAZYACCESS(modifiers, RIGHT_CLICK))
+		return
+	if(!isliving(target))
+		return //include whatever value you need to here
```

Comment:
I didn't mean include my comment here, I meant that if you need something like `COMPONENT_CANCEL_ATTACK_CHAIN` or something here that you should have it. At the very least, remove that comment. Sorry if that was confusing.

Reply:
This should just make the user do a normal attack_hand rather then canceling it